library/tests/integration/questions_router_test.rs

```diff
@@ -0,0 +1,325 @@
+use library::{
+    adapter::repositories::{
+        in_memory::question::QuestionInMemoryRepository,
+        postgres::question_db::QuestionDBRepository,
+    },
+    core::{
+        entities::question::{QuestionEntity, QuestionId},
+        ports::question::QuestionPort,
+    },
+    routes::Router,
+};
+use serde_json::Value;
+use std::{str::FromStr, sync::Arc};
+use warp::http::StatusCode;
+use warp::test::request;
+
+#[tokio::test]
+async fn questions_router_in_memory_test() {
+    // Set up an in-memory question port for testing
+    let question_port: Arc<dyn QuestionPort + Send + Sync> =
+        Arc::new(QuestionInMemoryRepository::default());
+
+    // Create a Router instance with the in-memory question port
+    let router = Router::new(question_port.clone());
+    let routers = router.routes();
+
+    let raw_question_id = String::from("question-1");
+    let question_id = QuestionId::from_str(&raw_question_id.clone()).unwrap();
+
+    let question = QuestionEntity {
+        id: question_id.clone(),
+        title: "How to write integration tests in Rust?".to_string(),
+        content: "I'm trying to write integration tests for a Warp router, but I'm not sure how to do it. Can someone help me?".to_string(),
+        tags: Some(vec!["rust".to_string(), "warp".to_string()]),
+    };
+
+    // Test POST /questions to create a new question
+    let post_resp = request()
+        .method("POST")
+        .path("/questions")
+        .json(&question)
+        .reply(&routers)
+        .await;
+
+    assert_eq!(
+        post_resp.status(),
+        StatusCode::OK,
+        "Failed to create a new question"
+    );
+
+    // Test GET /questions to get the created question
+    let get_resp = request()
+        .method("GET")
+        .path(format!("/questions/{}", &raw_question_id.clone()).as_str())
+        .reply(&routers)
+        .await;
+
+    assert_eq!(
+        get_resp.status(),
+        StatusCode::OK,
+        "Failed to get the created question"
+    );
+
+    // Deserialize the response body into a Value
+    let response_body: Value =
```

Comment:
let get_question: QuestionEntity =
        serde_json::from_slice(&get_resp.body()).expect("Failed to deserialize response body");